ui/app/src/components/BlueprintForm.tsx

```diff
@@ -120,8 +120,9 @@ function BlueprintForm(props: BlueprintFormProps) {
       setInputs({ [e.target.name]: e.target.checked, submitted: false });
     } else if (e.target.name === 'siteId') {
       const invalidSiteId = (e.target.value.startsWith('0') || e.target.value.startsWith('-') || e.target.value.startsWith('_'));
+      const siteId = e.target.value.replace(/[^a-zA-Z0-9-_]/g, '').replace(/_/g, '-').toLowerCase();
```

Comment:
the first regex is stripping off the underscores, so the second won't do anything

Reply:
that regex starts with `^`, so it will actually strip everything that is not `a-zA-Z0-9-_`